solr/core/src/java/org/apache/solr/parser/SolrQueryParserBase.java

```diff
@@ -1149,18 +1151,21 @@ protected Query getFieldQuery(String field, List<String> queryTerms, boolean raw
         return newFieldQuery
             (getAnalyzer(), field, queryText, false, fieldAutoGenPhraseQueries, fieldEnableGraphQueries, synonymQueryStyle);
       } else {
-        if (raw) {
+        if (raw) {// assumption: raw = false only when called from ExtendedDismaxQueryParser.getQuery()
           return new RawQuery(sf, queryTerms);
         } else {
           if (queryTerms.size() == 1) {
             return ft.getFieldQuery(parser, sf, queryTerms.get(0));
+          } else if(ft instanceof StrField){
```

Comment:
Rather than hard-code this specific field type, maybe you should call `ft.isTokenized()==false`?

Reply:
Unfortunately ft.isTokenized() is false also for numericals.
And for numericals sow=false doesn't need to change anything.
 It seems we need this behaviour with sow=false , only for StrField?

Reply:
In essence, I think the behavior I see here was correct *before* -- no special case for either StrField or numerics. In the context of the logic that reaches this point, the field is already ft.isTokenized==false.

Reply:
For numeric I agree with you, tokenization has little sense but for string fields, the won't be tokenized in the index.

Why would you like to have them tokenized at query time even if the user explicitly specified sow=false?

Reply:
Adding a special case that references a specific field type is a code smell.  Ideally, the FieldType API and/or interfaces express the logic to gate on here.  There are already too many special cases in the existing code IMO.  It seems to be already the case with ft.isTokenized().  I know the ultimate result is debatable but it's both-ways -- there's no obvious correct answer IMO.  The differing behavior is something users are advised to steer clear away from, e.g. by using consistently analyzed fields in "qf".

Reply:
Should we then start a much more complex task i.e. "removal of StrField" to avoid such inconsistencies?
We could try doing it in a back-compatible way, but ultimately I think the real issue is that we express text to be untokenized in the index in two ways with no apparent benefit but just inconsistencies ?